src/onboarding/registration/ProtectWallet.tsx

```diff
@@ -0,0 +1,241 @@
+import { NativeStackScreenProps } from '@react-navigation/native-stack'
+import React, { useLayoutEffect, useState } from 'react'
+import { useAsync } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { ActivityIndicator, View, ScrollView, StyleSheet, Text } from 'react-native'
+import { SafeAreaView } from 'react-native-safe-area-context'
+import { useSelector } from 'react-redux'
+import { OnboardingEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Button, { BtnSizes, BtnTypes } from 'src/components/Button'
+import { HeaderTitleWithSubtitle, nuxNavigationOptionsOnboarding } from 'src/navigator/Headers'
+import { navigate } from 'src/navigator/NavigationService'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import GuideKeyIcon from 'src/icons/GuideKeyIcon'
+import { getOnboardingStepValues, onboardingPropsSelector } from 'src/onboarding/steps'
+import { default as useTypedSelector } from 'src/redux/useSelector'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import OnboardingCard from 'src/components/OnboardingCard'
+import { twelveWordMnemonicEnabledSelector, walletAddressSelector } from 'src/web3/selectors'
+import seedrandom from 'seedrandom'
+import BottomSheet from 'src/components/BottomSheet'
+import { getWalletAsync } from 'src/web3/contracts'
+import { getPassword } from 'src/pincode/authentication'
+import { UnlockableWallet } from '@celo/wallet-base'
+import { UNLOCK_DURATION } from 'src/web3/consts'
+import variables from 'src/styles/variables'
+
+type Props = NativeStackScreenProps<StackParamList, Screens.ProtectWallet>
+
+function getCloudBackupIndex(seed: string | null) {
+  if (!seed) {
+    return 1
+  } else {
+    return seedrandom(seed)() > 0.5 ? 0 : 1
```

Comment:
`Math.random` doesn't allow you to provide a seed to get repeatable randomness, and I wasn't able to find anything easily accessible in libraries we currently used, so brought this in as a simple solution.

Reply:
How would that work? We'd have to split the fakedoor variant up into two separate ones, right? That would cause any effect to be more difficult to discern in Statsig, right? Unless I'm missing something here. Agreed that it would be preferable to not have this sort of logic in-code, though.

Reply:
yeah we'd end up with 2 fakedoor variants. I guess that would mean we'd get wider error bars on the metric for # of users clicking the fakedoor button. In total we'd need less manual analysis to break down by fakedoor button position, but more manual analysis to combine the fakedoor click statistic between the two fakedoor variants and compute an error bar. So maybe those two things cancel each other out. 

Reducing in-code logic could still be nice. Up to you.

Reply:
Here's the logic I wrote for bucketing an address for rolling out FC providers https://github.com/valora-inc/valora-rest-api/blob/2a64fefbc657804323466c4a571b9ccb38e4f2d1/src/fiatConnect/getProviderAvailability.ts#L34-L39